algo/src/main/java/com/hartwig/actin/algo/evaluation/general/HasMaximumWHOStatus.java

```diff
@@ -29,7 +32,17 @@ public Evaluation evaluate(@NotNull PatientRecord record) {
                     .build();
         }
 
-        if (who <= maximumWHO) {
+        Set<String> warningComplicationCategories = WHOFunctions.findComplicationCategoriesAffectingWHOStatus(record);
+
+        if (who == maximumWHO && !warningComplicationCategories.isEmpty()) {
+            return EvaluationFactory.unrecoverable()
+                    .result(EvaluationResult.WARN)
+                    .addWarnSpecificMessages(
+                            "Patient WHO status " + who + " equals maximum but patient has complication categories of concern: "
+                                    + Format.concat(warningComplicationCategories))
+                    .addWarnGeneralMessages("WHO adequate but has " + Format.concat(warningComplicationCategories))
+                    .build();
+        } else if (who <= maximumWHO) {
```

Comment:
I generally try to retain ordering "from good to bad" in evaluations, something like:
if (A) -> PASS, else if (B) -> WARN -> else FAIL

In this situation I think I would create a nested if statement: top one is "if who <= maximumWHO", and then inside that if statement evaluate complications, and do "if no dubious complications -> PASS, else WARN

Reply:
That ordering makes sense to me. I dealt with this WARN condition first here because it's a little harder to represent the PASS conditions on their own. For example, in a nested if statement, the first condition would have to be:
```
if (noDubiousComplications || who < maximumWHO) {
```
which means an additional check on `who`. Alternatively, we could handle `who < maximumWHO` in the first top-level condition, but then we have to factor out the duplicate `EvaluationResult` building. Would you still prefer to change this code to use the preferred ordering?

Reply:
Now I am doubting! I wouldn't consider the additional check on who to be bothersome, but more the fact that you need to invert the if "logical" if statement. We would indeed end up with your if statement and not sure it would make the code more readable. 

If you have no strong thoughts on this then maybe just let this be. I am pretty sure there are other rules that do not strictly follow PASS->WARN->FAIL, though it could still serve as our principle